apps/wallet-mobile/src/features/Receive/common/SingleOrMultipleAddressesModal/SingleOrMultipleAddressesModal.tsx

```diff
@@ -0,0 +1,97 @@
+import {useTheme} from '@yoroi/theme'
+import * as React from 'react'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {Button, Spacer, useModal} from '../../../../components'
+import {AddressMode} from '../../../../wallet-manager/types'
+import {useAddressModeManager} from '../../../../wallet-manager/useAddressModeManager'
+import {QRs} from '../../illustrations/QRs'
+import {useMultipleAddressesInfo} from '../useMultipleAddressesInfo'
+import {useStrings} from '../useStrings'
+
+export const singleOrMultipleAddressesModalHeight = 580
+
+type Props = {
+  onConfirm?: (method: AddressMode) => void
+}
+
+export const SingleOrMultipleAddressesModal = ({onConfirm}: Props) => {
+  const {styles, colors} = useStyles()
+  const strings = useStrings()
+  const {enableMultipleMode, enableSingleMode} = useAddressModeManager({
+    onSuccess: (_data, mode) => onConfirm?.(mode),
```

Comment:
since is on the higher tree, it can be just a effect, where the `mode` changes, after mounted.